compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaPathParameterSymbol.java

```diff
@@ -105,8 +116,15 @@ public TypeSymbol typeDescriptor() {
             return this.typeDescriptor;
         }
 
+        BType bType;
+        if (this.segmentKind == Kind.PATH_REST_PARAMETER && this.getInternalSymbol() instanceof BVarSymbol) {
```

Comment:
May be able to use SymbolKind instead of instanceof. Can check later/separately.

Reply:
Here we get `RESOURCE_PATH_REST_PARAM_SEGMENT` and hence we can't check whether it is a BVarSymbol or not